src/applications/disability-benefits/all-claims/components/ArrayField.jsx

```diff
@@ -0,0 +1,428 @@
+    });
+  }
+
+  /*
+   * Clicking Update on an item that’s not last and is in edit mode
+   */
+  handleUpdate(index) {
+    if (errorSchemaIsValid(this.props.errorSchema[index])) {
+      this.setState(_.set(['editing', index], false, this.state), () => {
+        this.scrollToTop();
+      });
+    } else {
+      // Set all the fields for this item as touched, so we show errors
+      const touched = setArrayRecordTouched(this.props.idSchema.$id, index);
+      this.props.formContext.setTouched(touched, () => {
+        scrollToFirstError();
+      });
+    }
+  }
+
+  /*
+   * Clicking Save
+   */
+  handleSave() {
+    const lastIndex = this.props.formData.length - 1;
+    if (errorSchemaIsValid(this.props.errorSchema[lastIndex])) {
+      // When we add another, we want to change the editing state of the currently
+      // last item, but not ones above it
+      const newEditing = this.state.editing.map(
+        (val, index) => (index + 1 === this.state.editing.length ? false : val),
+      );
+      const newState = _.assign(this.state, {
+        editing: newEditing,
+      });
+
+      this.setState(newState);
+    } else {
+      const touched = setArrayRecordTouched(this.props.idSchema.$id, lastIndex);
+      this.props.formContext.setTouched(touched, () => {
+        scrollToFirstError();
+      });
+    }
+  }
+
+  /*
+   * Clicking Add Another
+   */
+  handleAdd() {
+    const lastIndex = this.props.formData.length - 1;
+    if (errorSchemaIsValid(this.props.errorSchema[lastIndex])) {
+      // When we add another, we want to change the editing state of the currently
+      // last item, but not ones above it
+      const newEditing = this.state.editing.map(
+        (val, index) => (index + 1 === this.state.editing.length ? false : val),
+      );
+      const newState = _.assign(this.state, {
+        editing: newEditing.concat(true),
+      });
+
+      this.setState(newState, () => {
+        const newFormData = this.props.formData.concat(
+          getDefaultFormState(
+            this.props.schema.additionalItems,
+            undefined,
+            this.props.registry.definitions,
+          ) || {},
+        );
+        this.props.onChange(newFormData);
+        this.scrollToRow(`${this.props.idSchema.$id}_${lastIndex + 1}`);
+      });
+    } else {
+      const touched = setArrayRecordTouched(this.props.idSchema.$id, lastIndex);
+      this.props.formContext.setTouched(touched, () => {
+        scrollToFirstError();
+      });
+    }
+  }
+
+  /*
+   * Clicking Remove on an item in edit mode
+   */
+  handleRemove(indexToRemove) {
+    let newItems = this.props.formData.filter(
+      (val, index) => index !== indexToRemove,
+    );
+    let newState = _.assign(this.state, {
+      editing: this.state.editing.filter(
+        (val, index) => index !== indexToRemove,
+      ),
+    });
+    // Ensure that there is always a blank item in editing mode
+    if (newItems.length === 0) {
```

Comment:
I'm not sure why this wasn't working, but when I tried to delete the last item, it looks like it didn't update the state properly, but gave me a validation message:

![image](https://user-images.githubusercontent.com/12970166/52369536-ca5ff480-2a05-11e9-9f85-e22ad49b2990.png)

I think if we don't show the remove button on the last item, we won't have to have this condition in here, though.

Reply:
I'd vote for not showing a `Remove` button when there's only one item.

Reply:
I agree with getting rid of it; cleaner and makes more sense